Controllers/UsersController.cs

```diff
@@ -57,25 +57,32 @@ public async Task<IActionResult> Login(LoginDTO model)
             return ApiBadRequest("Bad password");
         }
 
-        [HttpPost("signup")]
-        [AllowAnonymous]
+       // [Authorize(Roles = "Admin")]
+        [HttpPost("register")]
         public async Task<IActionResult> Signup(RegisterDTO model)
         {
             if (!IsValidApiRequest())
             {
                 return ApiBadRequest("Invalid Headers!");
             }
 
-            var user = new Employee
+            var user = new Employee(model);
+
+            switch (model.RoleId)
             {
-                Username = model.Username,
-                Email = model.Email,
-                Department = model.RoleId,
-                FirstName = "Test",
-                LastName = "Testing",
-                PersonalCode = "39000000000",
-                Status = EmployeeStatusId.Employed
-            };
+                case DepartmentId.Pharmacy:
+                    user.Pharmacy = Context.Pharmacy.FirstOrDefault(z => z.Id == model.PharmacyWarehouseOrTruck);
+                    break;
+                case DepartmentId.Warehouse:
+                    user.Warehouse = Context.Warehouse.FirstOrDefault(z => z.Id == model.PharmacyWarehouseOrTruck);
+                    break;
+                case DepartmentId.Transportation:
+                    Context.TruckEmployees.Add(new TruckEmployee() {
+                        Truck = Context.Truck.FirstOrDefault(z => z.Id == model.PharmacyWarehouseOrTruck),
+                        Employee = user
+                    });
+                    break;
+            }
```

Comment:
Maybe it is better to put this logic below password validation (line 92)? Now we first start by creating object and relation and if the password os oncorrect, we just throw it all away. This is inefficient.